mkt/account/serializers.py

```diff
@@ -38,42 +35,38 @@ class Meta:
         model = UserProfile
         fields = ['source', 'verified']
 
-    def transform_source(self, obj, value):
+    def to_representation(self, obj):
         """Return the sources slug instead of the id."""
+        data = super(AccountInfoSerializer, self).to_representation(obj)
         if obj.pk is None:
-            return mkt.LOGIN_SOURCE_LOOKUP[mkt.LOGIN_SOURCE_UNKNOWN]
+            source = mkt.LOGIN_SOURCE_LOOKUP[mkt.LOGIN_SOURCE_UNKNOWN]
         elif obj.source in self.ALLOWED_SOURCES:
-            return mkt.LOGIN_SOURCE_LOOKUP[value]
+            source = mkt.LOGIN_SOURCE_LOOKUP[obj.source]
         else:
-            return mkt.LOGIN_SOURCE_LOOKUP[mkt.LOGIN_SOURCE_BROWSERID]
+            source = mkt.LOGIN_SOURCE_LOOKUP[mkt.LOGIN_SOURCE_BROWSERID]
+
+        data["source"] = source
+        return data
 
 
 class FeedbackSerializer(PotatoCaptchaSerializer):
-    feedback = fields.CharField()
-    platform = fields.CharField(required=False)
+    feedback = fields.CharField(allow_blank=False)
     chromeless = fields.CharField(required=False)
     from_url = fields.CharField(required=False)
-    user = fields.Field()
+    user = fields.ReadOnlyField(required=False)
+    platform = fields.CharField(required=False, allow_null=True)
 
     def validate(self, attrs):
         attrs = super(FeedbackSerializer, self).validate(attrs)
-
         if not attrs.get('platform'):
             attrs['platform'] = self.request.GET.get('dev', '')
         if self.request.user.is_authenticated():
-            attrs['user'] = self.request.user
+            attrs['user'] = unicode(self.request.user)
```

Comment:
This looked suspicious to me. I took a quick glance at the code and I don't understand why we'd need that at all, since the view uses the serialized data only to build a context dict for the email function, and overrides the `user` key when it does that.